Utilities/NanoAODReader.cc

```diff
@@ -121,7 +122,7 @@ class NanoAODReader {
     ReadTriggers();
     ReadVertices();
     RegionAssociations = RegionReader();
-    if(IsMC) CalcEventSFweights();
+    if(IsMC) EventWeights = CalcEventSFweights();
```

Comment:
This is to avoid building a neverending vector. Good catch of you.

Reply:
I'd suggest do EventWeights.clear(). Because currently your SFWeights is not reset after an event. Simply put EventWeights.clear(); SFWeights.clear() seems a better way for me.

Reply:
Sicne the function CalcEventSFweights() produces a vector of given size, just equaling this will reproduce the same size vector each event. There's not really a reason to make a global variable, clear it, then push back into it, each event, if instead I can just replace it outright with a new vector, which is implicitly what happens, as each time CalcEventSFweights is now called, it starts with a new empty vector and fills it, anyways.

Reply:
Alright. I thought creating vectors costs more run time than clearing them.